src/binary-reader-linker.cc

```diff
@@ -163,6 +163,11 @@ Result BinaryReaderLinker::OnImportGlobal(uint32_t import_index,
   return Result::Ok;
 }
 
+Result BinaryReaderLinker::OnFunctionCount(uint32_t count) {
+  binary->debug_names.resize(count + binary->function_imports.size());
```

Comment:
Oops! You're perfectly right, I was misreading OnFunctionCount as OnFunctionNamesCount. My apologies.